lib/matplotlib/colorbar.py

```diff
@@ -551,14 +554,22 @@ def update_normal(self, mappable):
             self.norm = mappable.norm
             self._reset_locator_formatter_scale()
 
-        self.draw_all()
+        self._draw_all()
         if isinstance(self.mappable, contour.ContourSet):
             CS = self.mappable
             if not CS.filled:
                 self.add_lines(CS)
         self.stale = True
 
+    @_api.deprecated("3.6", alternative="fig.draw_without_rendering()")
     def draw_all(self):
+        """
+        Calculate any free parameters based on the current cmap and norm,
+        and do all the drawing.
+        """
+        self._draw_all()
+
+    def _draw_all(self):
```

Comment:
Can we rename this to `_update_parameters()` or something like that instead? A quick search shows that the only other `draw_all()` is in pyplot to update all stale figures, which is very different from this `draw_all`.

As an aside, I'm wondering if this information should actually be in a `draw(self, renderer)` method instead (it looks like Colorbar doesn't have a draw() currently), that can then call `self.ax.draw()` after these updates.